### PR TITLE
[ADMIN] added past bans check cronjob

### PR DESCRIPTION
**Pull request by danielepantaleone**
Implemented past bans check: problem described here: http://forum.bigbrotherbot.net/general-usage-support/banning-with-multiple-servers-with-the-same-database-problem/msg44896/#msg44896

### CODE REVIEW

**Comment by thomasleveil** [comment]
while this PR does solves the issue I'd like your view on a different approach:

- B3 starts
- B3 query the db for the current bans count and save that number in its memory
- every 5s B3 query the database for the bans count and compare to its number in memory
- if and only if the counts are different : 
  - check online players and eventually kick some of them
  - update the ban count in the B3 memory to match the actual count in DB

The aim of this approach is:
- reduce the lag between a ban that occurs in Echelon and the actual kick on the game server
- reduce the load on the game server (no need to query the player list every so often)

**Comment by danielepantaleone** [comment]
I must say that I like your approach more. I would only increase those 5s to something like 10s, or make it configurable: in this way we do not hit the database too often.

**Comment by thomasleveil** [comment]
agreed this delay should be customizable. Note that the [`storage.getCounts`](https://github.com/BigBrotherBot/big-brother-bot/blob/release-1.10/b3/storage/common.py#L105) already provides the numbers to watch for change and those SQL queries are quite light on the database engine so short delays should not do any harm.

**Comment by danielepantaleone** [comment]
Yes, I was watching the same method but I was thinking of implementing a new one (which would be half duplicated): `getNumPenalties`. In this way we do not have to make 2 SQL queries since we do not care about the number of clients. I agree that it wouldn't impact a lot on the DBMS but since it's not that of a big deal implementing a new method, I would go for this option.

**Comment by danielepantaleone** [comment]
I rebased my index tree adding the changes suggested

File: b3/plugins/admin/__init__.py [hunk 1]
```diff
@@ -1032,6 +1060,21 @@ def doList(self, client, cmd):
         cmd.sayLoudOrPM(client, ', '.join(names))
         return True
 
+    def doPastBansCheck(self):
+        """
+        Scheduled execution: will check for banned players still
+        connected to the server and kick them if needed.
+        """
+        counts = self.console.storage.getCounts()
+        if counts['Bans'] + counts['TempBans'] != self._past_bans_count:
```
Comment (thomasleveil) [inline b3/plugins/admin/__init__.py]: This logic is flawn, take the following case for instance:

T0: 2 bans and 3 tempbans = 5 penalties
T1: 3 bans and 2 tempbans = 5 penalties

You do need to compare both counts separately

**Comment by danielepantaleone** [comment]
You are correct. I'll wait for this evening to have a bit of time to review my code. I never actually understoof why B3 consider Bans and TempBans separately. I mean, a ban is a ban and then it doesn't expire is permanent. I don't mean to change this behaviour, just out of curiosity :smile:

**Comment by danielepantaleone** [comment]
Alright, took 10 minutes off to fix this. I used some nested for loops instead of fixed dictionary keys to perform separate checks: this may sound a bit silly but in case we want to add "new" type of bans (can't think of another to be honest, but you never know), we just have to modify the class attribute holding the counts.

**Comment by thomasleveil** [comment]
LGTM thx a lot for your work :cake:

**Comment by thomasleveil** [comment]
available in [B3 daily build 276](http://files.cucurb.net/b3/daily/)

**Pull request merged**